docs/concepts/services.mdx

```diff
@@ -192,13 +188,90 @@ Virtual objects expose a set of handlers with access to K/V state stored in Rest
             ```diff focus=1,2,7,8,13,15,16,23
             ```
 
-            - **Concurrency guarantees**: to ensure consistent writes to the state, at most one handler can execute at a time for a given virtual object.
+            **Concurrency guarantees**: to ensure consistent writes to the state, at most one handler can execute at a time for a given virtual object.
             This can also be used, for example, to implement a locking mechanism or to ensure single writer to a database row.
 
         </CH.Spotlight>
     </TabItem>
 </Tabs>
 
+## Workflows
+
+A workflow is a special type of Virtual Object.
+
+
+
+<Tabs groupId="sdk" queryString>
+    <TabItem value="ts" label="TypeScript">
+
+        <CH.Spotlight className={"spotlight-long-code"}>
+
+            ```typescript
+            CODE_LOAD::ts/src/concepts/workflow.ts#<start_workflow>-<end_workflow>
+            ```
+
+            ---
+
+            ```diff focus=4:25
+            ```
+
+            A workflow has a **`run` handler** that implements the **workflow logic**.
+            The `run` handler runs exactly once per workflow ID (object).
+
+            ---
+
+            ```diff focus=16,22,31
+            ```
+
+            You can **query the workflow** by defining other handlers in the same object.
+            For example, you can store state in the workflow object, and query it from other handlers.
+            These handlers are able to run concurrently to the `run` handler.
+
+            ---
+
+            ```diff focus=15,27:29
+            ```
+
+            You can **signal the workflow**, to send information to it.
+            For example, the payment provider signals the workflow that the payment was successful by calling the `paymentWebhook`.
+
+        </CH.Spotlight>
+    </TabItem>
+    <TabItem value="java" label="Java">
```

Comment:
we don't have kotlin examples :(

Reply:
I know... It's a to do with slightly lower priority... We don't have kotlin anywhere in the concepts...